mhs/inbound/inbound/request/handler.py

```diff
@@ -33,14 +34,21 @@ def initialize(self, workflows: Dict[str, workflow.CommonWorkflow],
         self.party_id = party_id
         self.state_store = state_store
 
+    @time_request
     async def post(self):
-        logger.info('001', 'Inbound POST recieved: {request}', {'request': self.request})
+        logger.info('001', 'Inbound POST received: {request}', {'request': self.request})
+        request_message = None
+        try:
+            request_message = ebxml_request_envelope.EbxmlRequestEnvelope.from_string(self.request.headers,
+                                                                                      self.request.body.decode())
+        except ebxml_envelope.EbXmlParsingError as e:
+            logger.error('020', f'Failed to parse response: {e}')
+            raise tornado.web.HTTPError(500, 'Error occurred during message parsing',
+                                        reason=f'Exception in workflow') from e
```

Comment:
Saying _Exception in workflow_ is a little misleading here if someone's diagnosing the error.